File: Python/power_of_two/power_of_two.py
class Solution(object):
    def isPowerOfTwo(self, n):
        is_power = False
        if n <= 0:
            return is_power
        while n > 0:
            if n & 1:
                if is_power:
                    return False
                else:
                    is_power = not is_power
            n = n >> 1
        return True
        

File: Cpp/power_of_two/power_of_two.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        bool is_power = false;
        while (n > 0) {
            if (n & 1) {
                if (is_power == false) {
                    is_power = ! is_power;
                }
                else {
                    return false;
                }
            }
            n = n >> 1;
        }
        return is_power;
    }
};